examples/generative/cyclegan.py

```diff
@@ -0,0 +1,695 @@
+                discriminator_y_optimizer=self.model.disc_Y_optimizer,
+            )
+            self.ckpt_manager = tf.train.CheckpointManager(
+                ckpt, self.checkpoint_path, max_to_keep=5
+            )
+
+            if self.ckpt_manager.latest_checkpoint:
+                ckpt.restore(self.ckpt_manager.latest_checkpoint)
+                print("Latest checkpoint restored successfully!")
+        ckpt_save_path = self.ckpt_manager.save()
+        print(
+            "\nSaving checkpoint for epoch {} at {}".format(epoch + 1, ckpt_save_path)
+        )
+
+
+"""
+## Train the end-to-end model
+"""
+
+
+# Loss function for evaluating adverserial loss
+adv_loss_fn = keras.losses.MeanSquaredError()
+
+# Define the loss function for the generators
+def generator_loss_fn(fake):
+    fake_loss = adv_loss_fn(tf.ones_like(fake), fake)
+    return fake_loss
+
+
+# Define the loss function for the discriminators
+def discriminator_loss_fn(real, fake):
+    real_loss = adv_loss_fn(tf.ones_like(real), real)
+    fake_loss = adv_loss_fn(tf.zeros_like(fake), fake)
+    return (real_loss + fake_loss) * 0.5
+
+
+# Optimizers with parameters values as recommended
+gen_G_optimizer = keras.optimizers.Adam(learning_rate=2e-4, beta_1=0.5)
+gen_F_optimizer = keras.optimizers.Adam(learning_rate=2e-4, beta_1=0.5)
+disc_X_optimizer = keras.optimizers.Adam(learning_rate=2e-4, beta_1=0.5)
+disc_Y_optimizer = keras.optimizers.Adam(learning_rate=2e-4, beta_1=0.5)
+
+
+# Create cycle gan model
+cycle_gan_model = CycleGan(
+    generator_G=gen_G, generator_F=gen_F, discriminator_X=disc_X, discriminator_Y=disc_Y
+)
+
+# Compile the model
+cycle_gan_model.compile(
+    gen_G_optimizer=gen_G_optimizer,
+    gen_F_optimizer=gen_F_optimizer,
+    disc_X_optimizer=disc_X_optimizer,
+    disc_Y_optimizer=disc_Y_optimizer,
+    gen_loss_fn=generator_loss_fn,
+    disc_loss_fn=discriminator_loss_fn,
+)
+# Callbacks
+plotter = GANMonitor()
+checkpointer = ModelCheckpointer(checkpoint_path="./checkpoints_cyclegan/")
+
+# Here we will train the model for just one epoch as each epoch takes around
+# 7 minutes on a single P100 backed machine.
+cycle_gan_model.fit(
+    tf.data.Dataset.zip((train_horses, train_zebras)),
+    epochs=1,
+    callbacks=[plotter, checkpointer],
+)
+
+"""
+Test the performance of the model.
+"""
+
+
+# The same model was trained for 112 epochs. We will be loading those weights
+# here. Once the weights are loaded, we will take a few samples from the test
+# data and check the model's performance.
+
+ckpt = tf.train.Checkpoint(
+    generator_g=gen_G,
+    generator_f=gen_F,
+    discriminator_x=disc_X,
+    discriminator_y=disc_Y,
+    generator_g_optimizer=gen_G_optimizer,
+    generator_f_optimizer=gen_F_optimizer,
+    discriminator_x_optimizer=disc_X_optimizer,
+    discriminator_y_optimizer=disc_Y_optimizer,
+)
+
+ckpt.restore("./checkpoints_cyclegan/ckpt-112").expect_partial()
```

Comment:
This will not be loadable when we execute the notebook. As a result the plot below won't be displayed. Could you upload this file somewhere and download it before loading it here?

Reply:
Yes, I am aware of that. I wanted to get the review of the py file first. I will upload the weights in a repo and we will load the weights here.